fmt/src/formatter.rs

```diff
@@ -514,6 +533,39 @@ impl<'a, W: Write> Visitor for Formatter<'a, W> {
                 }
                 Type::Function { .. } => self.visit_source(loc)?,
             },
+            Expression::ArraySubscript(_, ty_exp, size_exp) => {
+                ty_exp.visit(self)?;
+                write!(self, "[")?;
+                if let Some(size_exp) = size_exp {
+                    size_exp.visit(self)?;
+                }
+                write!(self, "]")?;
+            }
+            // Expression::Assign(_, var_expr, value_expr) => {
+            //     // TODO handle with operators?
```

Comment:
what do we need to handle here?

Reply:
I'll get rid of this section as per my comment above for right now

Reply:
there are a lot of expressions we need to handle here, but they basically get split up into a few categories, binary operators, unary operators and literal expresions and a bunch of one off stuff like assigning, identifiers, member access, function calls etc. My plan was to start with binary operators and build up the correct parentheses management / operator precedence, indentation etc after I get comments working